pulsar-transaction/coordinator/src/test/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionLogImplTest.java

```diff
@@ -0,0 +1,91 @@
+package org.apache.pulsar.transaction.coordinator.impl;
+
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.bookkeeper.mledger.ManagedLedgerConfig;
+import org.apache.pulsar.common.api.proto.Subscription;
+import org.apache.pulsar.transaction.coordinator.TransactionCoordinatorID;
+import org.apache.pulsar.transaction.coordinator.TransactionRecoverTracker;
+import org.apache.pulsar.transaction.coordinator.TransactionTimeoutTracker;
+import org.apache.pulsar.transaction.coordinator.proto.TransactionMetadataEntry;
+import org.apache.pulsar.transaction.coordinator.test.MockedBookKeeperTestCase;
+import static org.mockito.Mockito.*;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class MLTransactionLogImplTest extends MockedBookKeeperTestCase {
+
+    @DataProvider(name = "variedBufferedWriteConfigProvider")
+    private Object[][] variedBufferedWriteConfigProvider(){
+        Object[][] args = new Object[4][];
+        args[0] = new Object[]{true, true};
+        args[1] = new Object[]{false, false};
+        args[2] = new Object[]{true, false};
+        args[3] = new Object[]{false, true};
+        return args;
+    }
+
+    @Test(dataProvider = "variedBufferedWriteConfigProvider")
+    public void test1(boolean writeWithBatch, boolean readWithBatch) throws Exception {
```

Comment:
Please provide a meaningful test name.

Reply:
And looks like you have created a data provider, but both of the parameters are not used in the test.

Reply:
> Please provide a meaningful test name.

Already fixed.

> And looks like you have created a data provider, but both of the parameters are not used in the test.

Already fixed.